harness/simdUtilities.js

```diff
@@ -0,0 +1,449 @@
+  signed: true,
+  numerical: true,
+  logical: true,
+  lanes: 16,
+  laneSize: 1,
+  laneMask: 0xFF,
+  minVal: -0x80,
+  maxVal: 0x7F,
+  interestingValues: [0, 1, -1, 0x40, 0x7F, -0x80],
+  view: Int8Array,
+  buffer: _i8x16,
+  mulFn: binaryMul,
+}
+
+var uint32x4 = {
+  name: "Uint32x4",
+  fn: SIMD.Uint32x4,
+  intLane: true,
+  unsigned: true,
+  numerical: true,
+  logical: true,
+  lanes: 4,
+  laneSize: 4,
+  minVal: 0,
+  maxVal: 0xFFFFFFFF,
+  interestingValues: [0, 1, 0x40000000, 0x7FFFFFFF, 0xFFFFFFFF],
+  view: Uint32Array,
+  buffer: _ui32x4,
+  mulFn: binaryImul,
+}
+
+var uint16x8 = {
+  name: "Uint16x8",
+  fn: SIMD.Uint16x8,
+  intLane: true,
+  unsigned: true,
+  numerical: true,
+  logical: true,
+  lanes: 8,
+  laneSize: 2,
+  laneMask: 0xFFFF,
+  minVal: 0,
+  maxVal: 0xFFFF,
+  interestingValues: [0, 1, 0x4000, 0x7FFF, 0xFFFF],
+  view: Uint16Array,
+  buffer: _ui16x8,
+  mulFn: binaryMul,
+}
+
+var uint8x16 = {
+  name: "Uint8x16",
+  fn: SIMD.Uint8x16,
+  intLane: true,
+  unsigned: true,
+  numerical: true,
+  logical: true,
+  lanes: 16,
+  laneSize: 1,
+  laneMask: 0xFF,
+  minVal: 0,
+  maxVal: 0xFF,
+  interestingValues: [0, 1, 0x40, 0x7F, 0xFF],
+  view: Int8Array,
+  buffer: _ui8x16,
+  mulFn: binaryMul,
+}
+
+var bool32x4 = {
+  name: "Bool32x4",
+  fn: SIMD.Bool32x4,
+  boolLane: true,
+  lanes: 4,
+  laneSize: 4,
+  interestingValues: [true, false],
+}
+
+var bool16x8 = {
+  name: "Bool16x8",
+  fn: SIMD.Bool16x8,
+  boolLane: true,
+  lanes: 8,
+  laneSize: 2,
+  interestingValues: [true, false],
+}
+
+var bool8x16 = {
+  name: "Bool8x16",
+  fn: SIMD.Bool8x16,
+  boolLane: true,
+  lanes: 16,
+  laneSize: 1,
+  interestingValues: [true, false],
+}
+
+// Filter functions.
+function isFloatType(type) { return type.floatLane; }
+function isIntType(type) { return type.intLane; }
+function isBoolType(type) { return type.boolLane; }
+function isNumerical(type) { return type.numerical; }
+function isLogical(type) { return type.logical; }
+function isSigned(type) { return type.signed; }
+function isSignedIntType(type) { return type.intLane && type.signed; }
+function isUnsignedIntType(type) { return type.intLane && type.unsigned; }
+function isSmallIntType(type) { return type.intLane && type.lanes >= 8; }
+function isSmallUnsignedIntType(type) { return type.intLane && type.unsigned && type.lanes >= 8; }
+function hasLoadStore123(type) { return !type.boolLane && type.lanes == 4; }
+
+// Each SIMD type has a corresponding Boolean SIMD type, which is returned by
+// relational ops.
+float32x4.boolType = int32x4.boolType = uint32x4.boolType = bool32x4;
+int16x8.boolType = uint16x8.boolType = bool16x8;
+int8x16.boolType = uint8x16.boolType = bool8x16;
+
+// SIMD fromTIMD types.
+float32x4.from = [int32x4, uint32x4];
+int32x4.from = [float32x4, uint32x4];
+int16x8.from = [uint16x8];
+int8x16.from = [uint8x16];
+uint32x4.from = [float32x4, int32x4];
+uint16x8.from = [int16x8];
+uint8x16.from = [int8x16];
+
```

Comment:
If the spec has changed, we should update the tests. Otherwise, this would give false negatives, causing the people think that an engine's implementation is wrong. You can keep the changes as a separate Git commit if you'd like.